corehq/motech/repeaters/expression/repeater_generators.py

```diff
@@ -16,15 +16,16 @@ def content_type(self):
         return 'application/json'
 
     def get_payload(self, repeat_record, payload_doc, parsed_expression):
-        result = parsed_expression(payload_doc, EvaluationContext(payload_doc))
+        result = _generate_payload(payload_doc, parsed_expression)
         return json.dumps(result, cls=DjangoJSONEncoder)
 
     def get_url(self, repeat_record, url_template, payload_doc):
         if not toggles.UCR_EXPRESSION_REGISTRY.enabled(repeat_record.domain):
             return ""
 
         required_template_vars = [fn for _, fn, _, _ in Formatter().parse(url_template) if fn is not None]
-        context = EvaluationContext(payload_doc)
+        payload_doc_json = payload_doc.to_json()
+        context = EvaluationContext(payload_doc_json)
```

Comment:
This was broken for Form repeaters since the evaluation context would be created with an `XFormInstance` instead of the serialized form.